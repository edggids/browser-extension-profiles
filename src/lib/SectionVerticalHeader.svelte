<script lang="ts">
let className = "";
export {className as class};

export let title = "";
export let compact = false;

const paddings = compact ? "pt-4 pb-5" : "pt-6 pb-2.75";
</script>

<section class={`relative ${className}`}>
    <span data-testid="title" class="vertical-header text-xs uppercase transition-color text-current">{title}</span>

    <div data-testid="contentWrapper" class={`container transition-bg bg-current-light ${paddings}`}>
        <slot></slot>
    </div>
</section>
