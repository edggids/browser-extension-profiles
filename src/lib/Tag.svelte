<script lang="ts">
  import {createEventDispatcher} from "svelte";

  let className = '';
  export { className as class };

  export let active = false;

  export let color = '#1d1e1e';

  const dispatch = createEventDispatcher();

  function onClick() { dispatch('click'); }
</script>

<span on:click={onClick} class:active class={ `tag transition-border-color transition-color rounded-xl p-2.75 leading-none pr-4 inline-flex items-center text-sm bg-ground ${className}` } style:--color={color}>
  <span class="indicator mr-2.75"></span><slot></slot>
</span>
