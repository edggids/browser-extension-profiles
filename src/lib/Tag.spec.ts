import {cleanup, fireEvent, render} from '@testing-library/svelte';
import {describe, it, expect, afterEach, vi} from 'vitest';
import Tag from "./Tag.svelte";
import SectionVerticalHeader from "./SectionVerticalHeader.svelte";

describe('Tag component', () => {
  afterEach(() => {
    cleanup();
  });

  it('mounts', () => {
    const { container } = render(Tag, { props: { color: '#123456' } });

    expect(container).toMatchSnapshot();
  });

  it('renders with provided color', () => {
    const { container } = render(Tag, { props: { color: '#123456' } });
    const tag = container.querySelector('span');
    expect(tag.style.getPropertyValue('--color')).toBe('#123456');
  });

  it('renders as non-active when active is false', () => {
    const { container } = render(Tag, { props: { active: false } });
    const tag = container.querySelector('span');
    expect(tag.classList.contains('active')).toBe(false);
  });

  it('renders as active when active is true', () => {
    const { container } = render(Tag, { props: { active: true } });
    const tag = container.querySelector('span');
    expect(tag.classList.contains('active')).toBe(true);
  });

  it('emits click event when clicked', async () => {
    const { container, component } = render(Tag);
    const tag = container.querySelector('span');

    const onClick = vi.fn();
    component.$on('click', onClick);

    await fireEvent.click(tag);
    expect(onClick).toHaveBeenCalled();
  });
});
