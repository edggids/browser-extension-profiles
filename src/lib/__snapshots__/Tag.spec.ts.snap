// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tag component > mounts 1`] = `
<body>
  <div>
    <span
      class="tag transition-border-color transition-color rounded-xl p-2.75 leading-none pr-4 inline-flex items-center text-sm bg-ground"
      style="--color: #123456;"
    >
      <span
        class="indicator mr-2.75"
      />
    </span>
  </div>
</body>
`;
